src/platform/unix.rs

```diff
@@ -1,37 +1,49 @@
 #![cfg(any(target_os = "linux", target_os = "dragonfly", target_os = "freebsd", target_os = "netbsd", target_os = "openbsd"))]
 
-use std::{os::raw, ptr, sync::Arc};
+use std::os::raw;
 
+#[cfg(feature = "x11")]
+use std::{ptr, sync::Arc};
+
+#[cfg(feature = "wayland")]
 use smithay_client_toolkit::window::{ButtonState as SCTKButtonState, Theme as SCTKTheme};
 
 use crate::{
-    dpi::Size,
     event_loop::{EventLoop, EventLoopWindowTarget},
     monitor::MonitorHandle,
     window::{Window, WindowBuilder},
 };
 
+#[cfg(feature = "x11")]
+use crate::dpi::Size;
+
+#[cfg(feature = "x11")]
+use crate::platform_impl::x11::{ffi::XVisualInfo, XConnection};
 use crate::platform_impl::{
-    x11::{ffi::XVisualInfo, XConnection},
     EventLoop as LinuxEventLoop, EventLoopWindowTarget as LinuxEventLoopWindowTarget,
     Window as LinuxWindow,
 };
 
 // TODO: stupid hack so that glutin can do its work
+#[cfg(feature = "x11")]
 #[doc(hidden)]
 pub use crate::platform_impl::x11;
 
+#[cfg(feature = "x11")]
 pub use crate::platform_impl::{x11::util::WindowType as XWindowType, XNotSupported};
 
 /// Additional methods on `EventLoopWindowTarget` that are specific to Unix.
 pub trait EventLoopWindowTargetExtUnix {
     /// True if the `EventLoopWindowTarget` uses Wayland.
+    #[cfg(feature = "wayland")]
     fn is_wayland(&self) -> bool;
     ///
```

Comment:
While we're at it, could you remove this `///` thing and just have a blank line in between.